### PR TITLE
feature: base shopping cart EventSourcedEntity on codegen

### PR DESCRIPTION
**Pull request by octonato**
Draft because on top of #10.

Ready for review otherwise.

### CODE REVIEW

File: samples/java-eventsourced-shopping-cart/src/main/java/eventsourced/shoppingcart/Main.java [hunk 1]
```diff
@@ -1,38 +0,0 @@
...
-          .registerEventSourcedEntity(
-              ShoppingCartEntity.class,
-              ShoppingCartApi.getDescriptor().findServiceByName("ShoppingCartService"),
-              com.example.shoppingcart.domain.ShoppingCartDomain.getDescriptor());
-
-  public static final void main(String[] args) throws Exception {
-    LOG.info("started");
-    SERVICE.start().toCompletableFuture().get();
-  }
-}
```
Comment (pvlugter) [inline samples/java-eventsourced-shopping-cart/src/main/java/eventsourced/shoppingcart/Main.java:38]: Intended to delete the sources for this sample?

File: samples/java-eventsourced-shopping-cart/src/main/proto/shoppingcart/shoppingcart_api.proto [hunk 2]
```diff
@@ -25,55 +25,54 @@ import "google/api/annotations.proto";
 import "google/protobuf/empty.proto";
 
 message AddLineItem {
-  string cart_id = 1 [(akkaserverless.field).entity_key = true];
-  string product_id = 2;
-  string name = 3;
-  int32 quantity = 4;
+    string cart_id = 1 [(akkaserverless.field).entity_key = true];
```
Comment (pvlugter) [inline samples/java-eventsourced-shopping-cart/src/main/proto/shoppingcart/shoppingcart_api.proto:28]: Not sure on your editor setup, but the recommended style for protobuf is indent of 2 spaces:

https://developers.google.com/protocol-buffers/docs/style

**Review by pvlugter — COMMENTED** [review]
Not sure this PR is committed correctly with the deleted sources. Would also be good to leave the tidier formatting for protobufs.

File: samples/java-eventsourced-shopping-cart/src/main/proto/shoppingcart/shoppingcart_api.proto (hunk 2 above)
Comment (octonato) [reply samples/java-eventsourced-shopping-cart/src/main/proto/shoppingcart/shoppingcart_api.proto:28]: hmm, that's probably badly configured and I probably did the same on the value entity PR.

**Comment by octonato** [comment]
> Not sure this PR is committed correctly with the deleted sources

I messed up with it. The sources were supposed to be moved, not deleted.

File: samples/java-eventsourced-shopping-cart/src/main/proto/shoppingcart/shoppingcart_api.proto [hunk 3]
```diff
@@ -56,24 +56,23 @@ service ShoppingCartService {
...
 
-  rpc AddItem(AddLineItem) returns (google.protobuf.Empty) {
+  rpc AddItem (AddLineItem) returns (google.protobuf.Empty) {
     option (google.api.http) = {
-      post: "/cart/{cart_id}/items/add"
-      body: "*"
-    };
+            post: "/cart/{cart_id}/items/add"
+            body: "*"
+        };
```
Comment (ennru) [inline samples/java-eventsourced-shopping-cart/src/main/proto/shoppingcart/shoppingcart_api.proto:63]: Still some non-standard code formatting.

File: samples/java-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/domain/ShoppingCartImpl.java [hunk 4]
```diff
@@ -14,69 +14,53 @@
...
 import java.util.stream.Collectors;
 
-/** An event sourced entity. */
+/**
+ * An event sourced entity.
+ */
+
+/**
+ * An event sourced entity.
+ */
```
Comment (ennru) [inline samples/java-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/domain/ShoppingCartImpl.java:35]: Useless comments become better if you have them duplicated!

**Review by ennru — APPROVED** [review]
LGTM.

File: samples/java-eventsourced-shopping-cart/src/main/proto/shoppingcart/shoppingcart_api.proto (hunk 3 above)
Comment (octonato) [reply samples/java-eventsourced-shopping-cart/src/main/proto/shoppingcart/shoppingcart_api.proto:63]: yeah, the intellij formatter don't handle annotation properly.
I will fix by hand.

File: samples/java-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/domain/ShoppingCartImpl.java (hunk 4 above)
Comment (octonato) [reply samples/java-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/domain/ShoppingCartImpl.java:35]: I'm glad you noticed! :-)

**Pull request merged**